examples/chat/client/tasks.nim

```diff
@@ -19,52 +19,65 @@ type
 
   StatusState* = enum loggedout, loggingin, loggedin, loggingout
 
+  WakuState* = enum stopped, starting, started, stopping
+
+const DefaultTopic = waku_chat2.DefaultTopic
+
 var
   conf {.threadvar.}: ChatConfig
   connected {.threadvar.}: bool
-  contentTopic {.threadvar.}: ContentTopic
+  contentTopics {.threadvar.}: OrderedSet[ContentTopic]
   contextArg {.threadvar.}: StatusArg
+  extIp {.threadvar.}: Option[ValidIpAddress]
+  extTcpPort {.threadvar.}: Option[Port]
+  extUdpPort {.threadvar.}: Option[Port]
+  natIsSetup {.threadvar.}: bool
   nick {.threadvar.}: string
+  nodekey {.threadvar.}: waku_chat2.crypto.PrivateKey
+  nodekeyGenerated {.threadvar.}: bool
   subscribed {.threadvar.}: bool
   status {.threadvar.}: StatusObject
   statusState {.threadvar.}: StatusState
   wakuNode {.threadvar.}: WakuNode
   wakuState {.threadvar.}: WakuState
 
-when PayloadV1:
-  var
-    symKey {.threadvar.}: SymKey
-    symKeyGenerated {.threadvar.}: bool
-
 proc resetContext() {.gcsafe, nimcall.} =
   connected = false
   nick = ""
   subscribed = false
   wakuNode = nil
   wakuState = WakuState.stopped
 
-const statusContext*: Context = proc(arg: ContextArg) {.async, gcsafe, nimcall,
+proc statusContext*(arg: ContextArg) {.async, gcsafe, nimcall,
   raises: [Defect].} =
 
-  # set threadvar values that are never reset, i.e. persist across dis/connect
+  # set threadvar values that are never reset, i.e. persist across
+  # waku chat2 dis/connect
   contextArg = cast[StatusArg](arg)
   conf = contextArg.chatConfig
-  contentTopic = conf.contentTopic
+
+  let contentTopicsStr = conf.contentTopics.strip()
+  if contentTopicsStr != "":
+    contentTopics = contentTopicsStr.split(" ").toOrderedSet()
```

Comment:
Shouldn't we already have this available to us in `contextArg.topics`, but as an `OrderedSet[string]`? It feels like maybe the `ContextArg` type should have `OrderedSet[ContentTopic]`? But if you don't agree, then here we could simply convert from string to `ContentTopic` by way of `map`:

```nim
import sugar

contentTopics = contextArg.topics.map(t => ContentTopic t)
```

Actually, seeing that `ContentTopic` is not a distinct type of string, then likely we don't need a type conversion and can simply do:
```nim
contentTopics = contextArg.topics
```

Reply:
`ContentTopic` is effectively just an alias for Nim's `string`.

The threadvar here is setup as an `OrderedSet` instance for efficiency's sake.

Reply:
Understood. More to the point of the comment was that should we consider simply using the topics already parsed in to an `OrderedSet` from the `contextArg`? Like this:
```suggestion
 contentTopics = contextArg.topics
```